proxyd/server.go

```diff
@@ -421,6 +424,90 @@ func (s *Server) handleBatchRPC(ctx context.Context, reqs []json.RawMessage, isL
+				continue
+			}
+			var toAddr string
+			if err := json.Unmarshal(callObj["to"], &toAddr); err != nil {
+				log.Debug("error unmarshalling to address", "err", err, "req_id", GetReqID(ctx))
+				responses[i] = NewRPCErrorRes(parsedReq.ID, ErrInvalidParams("invalid to address"))
+				continue
+			}
+
+			// Check for the specific contract address
+			if strings.EqualFold(toAddr, "0xcA11bde05977b3631167028862bE2a173976CA11") {
+				// Check if data field exists
+				if callObj["data"] == nil {
+					log.Debug("missing data field in call object", "req_id", GetReqID(ctx))
+					responses[i] = NewRPCErrorRes(parsedReq.ID, ErrInvalidParams("missing data field"))
+					continue
+				}
+
+				var dataStr string
+				if err := json.Unmarshal(callObj["data"], &dataStr); err != nil {
+					log.Debug("error unmarshalling data field", "err", err, "req_id", GetReqID(ctx))
+					responses[i] = NewRPCErrorRes(parsedReq.ID, ErrInvalidParams("invalid data field"))
+					continue
+				}
+
+				// Map of data values to responses
+				responseMap := map[string]string{
+					// cUSD
+					"a54b7fc0000000000000000000000000765de816845861e75a25fca122bb6898b8b1282a": "000000000000000000000000000000000000000000000000000000024ab9f46a",
+					// USDC
+					"a54b7fc00000000000000000000000002f25deb3848c207fc8e0c34035b3ba7fc157602b": "000000000000000000000000000000000000000000000000000000024ab9f46a",
+					// cCOP
+					"a54b7fc00000000000000000000000008a567e2ae79ca692bd748ab832081c45de4041ea": "000000000000000000000000000000000000000000000000000024c3929fa684",
+					// USDT
+					"a54b7fc00000000000000000000000000e2a3e05bc9a16f5292a6170456a710cb89c6f72": "000000000000000000000000000000000000000000000000000000024ab9f46a",
+					// cEUR
+					"a54b7fc0000000000000000000000000d8763cba276a3738e6de85b4b3bf5fded6d6ca73": "000000000000000000000000000000000000000000000000000000021fb97d29",
+					// cREAL
+					"a54b7fc0000000000000000000000000e8537a3d056da446677b9e9d6c5db704eaab4787": "0000000000000000000000000000000000000000000000000000000d1b210737",
+					// eXOF
+					"a54b7fc000000000000000000000000073f93dcc49cb8a239e2032663e9475dd5ef29a08": "0000000000000000000000000000000000000000000000000000057076e4459b",
+					// cKES
+					"a54b7fc0000000000000000000000000456a3d042c0dbd3db53d5489e98dfb038553b0d0": "0000000000000000000000000000000000000000000000000000012908aa4e00",
+				}
+
+				var fixedResult string
+
+				// Check for partial matches
+				for pattern, response := range responseMap {
+					if strings.Contains(dataStr, pattern) {
+						fixedResult = "0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000020" + response
```

Comment:
How was this prefix calculated?

Reply:
Taken from the example at https://clabsco.slack.com/archives/CNJ7KTHQU/p1742982821704169?thread_ts=1742975149.704359&cid=CNJ7KTHQU and the specific token removed.